server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageBlobsDAO.java

```diff
@@ -92,7 +92,16 @@
 
     @Override
     public CompletableFuture<BlobId> save(byte[] data) {
-        return save(new ByteArrayInputStream(data));
+        HashingInputStream hashingInputStream = new HashingInputStream(Hashing.sha256(), new ByteArrayInputStream(data));
```

Comment:
you don't need all this code.
`Hashing.sha256().hashBytes(data)` do what you want